### PR TITLE
chore: updating test coverage for sink submodule

### PR DESCRIPTION
**Pull request by drduhe**
**Issue #, if available:** n/a

### Notes
This PR introduces comprehensive unit tests and enhancements across multiple sink-related classes, including `KinesisSink`, `S3Sink`, `Sink`, and `SinkFactory`. These changes aim to improve test coverage, consistency, and robustness of the sink implementations.

### `KinesisSink`
- **Expanded Test Coverage**: Added new test cases to handle:
  - Edge cases like empty feature lists.
  - Validation failures, oversized record handling, and partial write scenarios.
  - Correct behavior when using assumed role credentials.
- **Improved Test Clarity**: Added detailed docstrings to explain test scenarios.
- **Cleaned Up Mocks**: Ensured proper setup and teardown of stubs and mocks.

### `S3Sink`
- **New Test Scenarios**: Introduced tests to cover:
  - Handling slashes in `image_id` and validating correct file paths.
  - Multipart upload configurations for large data files.
  - Bucket permission issues (403 and 404 errors).
  - Writing empty feature lists to validate no operation occurs.
- **Enhanced Readability**: Refactored to use consistent mocking patterns and clear descriptions.

### `Sink`
- **Abstract Class Mock Implementation**: Created `MockSink` to enable testing of the `Sink` interface.
- **Basic Unit Tests**: Added tests for `name`, `mode`, `write`, and `__str__` methods to verify expected interface behavior.

### `SinkFactory`
- **New Tests for Sink Creation**:
  - Validated sink selection based on input configurations (S3 and Kinesis).
  - Handled invalid sink types gracefully, ensuring exceptions are raised.
  - Tested mixed sink setups (partial success/failure) to ensure correct results are logged and returned.
- **Error Handling**: Added scenarios to validate responses when no outputs are defined.

## Additional Improvements
- **Docstrings and Comments**: Enhanced all unit test files with clear, concise comments explaining the purpose and expected behavior of each test.
- **Consistent Naming and Structure**: Ensured uniformity across all test files for easier maintenance.

### Checklist

Before you submit a pull request, please make sure you have the following:
- [x] Code changes are compact and well-structured to facilitate easy review
- [x] Changes are documented in the README.md and other relevant documentation pages
- [x] PR title and description accurately reflect the changes and are detailed enough for historical tracking
- [x] PR contains tests that cover all new code and the code has been manual tested
- [x] All new dependencies are declared (if any), and no unnecessary libraries are added
- [x] Performance impacts (if any) of the changes are evaluated and documented
- [x] Security implications of the changes (if any) are reviewed and addressed
- [x] I have read the [Contributing Guidelines](../CONTRIBUTING.md) and agree to follow the [Code of Conduct](../CODE_OF_CONDUCT.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/aws/osml/model_runner/sink/sink_factory.py [hunk 1]
```diff
@@ -12,7 +13,7 @@
 logger = logging.getLogger(__name__)
 
 
-class SinkFactory(Sink):
+class SinkFactory(Sink, ABC):
```
Comment (edparris) [inline src/aws/osml/model_runner/sink/sink_factory.py:16]: SinkFactory should not extend Sink or ABC. It only needs to extend ABC if it will implement abstract methods that must be implemented by subclasses to use the factory. It shouldn't implement Sink because it isn't actually providing the name(), mode(), and write(), functions that define the Sink interface. I realize that extending Sink was in the base code not something you added but this change drew my attention to it.

File: test/aws/osml/model_runner/sink/test_sink.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        """
+        self.sink = MockSink()
+
+    def tearDown(self):
+        """
+        Clean up any resources used by the tests.
+        """
+        self.sink = None
+
+    def test_sink_name(self):
```
Comment (edparris) [inline test/aws/osml/model_runner/sink/test_sink.py:42]: Testing name, mode, and write on your mock really isn't testing anything. I can see the value in testing your __str__ representation since that is actual custom code but the rest of it is just code for coverage sake.

File: src/aws/osml/model_runner/sink/sink_factory.py (hunk 1 above)
Comment (drduhe) [reply src/aws/osml/model_runner/sink/sink_factory.py]: I have removed the extension of ABC and Sink in the base class.

File: test/aws/osml/model_runner/sink/test_sink.py (hunk 2 above)
Comment (drduhe) [reply test/aws/osml/model_runner/sink/test_sink.py:42]: Removed the other tests and left str representation
